lib/BC/SleighLifter.cpp

```diff
@@ -0,0 +1,1292 @@
+               *ptr_add = bldr.CreateAdd(*lifted_addr, offset);
+          return this->LiftStoreIntoOutParam(bldr, ptr_add, outvar);
+        }
+        break;
+      }
+      case OpCode::CPUI_PTRSUB: {
+        auto lifted_addr = this->LiftInParam(
+                 bldr, param0, this->insn_lifter_parent.GetWordType()),
+             lifted_offset = this->LiftIntegerInParam(bldr, param1);
+        if (lifted_addr.has_value() && lifted_offset.has_value()) {
+          return this->LiftStoreIntoOutParam(
+              bldr, bldr.CreateAdd(*lifted_addr, *lifted_offset), outvar);
+        }
+        break;
+      }
+      default: break;
+    }
+
+    return LiftStatus::kLiftedUnsupportedInstruction;
+  }
+
+  LiftStatus LiftVariadicOp(llvm::IRBuilder<> &bldr, OpCode opc,
+                            VarnodeData *outvar, VarnodeData *vars,
+                            int4 isize) {
+    switch (opc) {
+      // We shouldnt encounter this afaik MULTIEQUAL is a decompiler concept?
+      case OpCode::CPUI_MULTIEQUAL: {
+        llvm::Type *phi_type =
+            llvm::IntegerType::get(this->context, vars[0].size * 8);
+        llvm::PHINode *phi_node = bldr.CreatePHI(phi_type, isize);
+        for (int4 i = 0; i < isize; ++i) {
+          VarnodeData &var = vars[i];
+          auto inval = this->LiftInParam(
+              bldr, var, llvm::IntegerType::get(this->context, var.size * 8));
+          if (!inval.has_value()) {
+            return LiftStatus::kLiftedUnsupportedInstruction;
+          }
+          // TODO(alex): This isn't right, just using the current block to get things building.
+          // We need to track the incoming basic blocks for each value.
+          phi_node->addIncoming(*inval, bldr.GetInsertBlock());
+        }
+        return this->LiftStoreIntoOutParam(bldr, phi_node, outvar);
+      }
+      case OpCode::CPUI_CPOOLREF: {
+        // NOTE(alex): We shouldn't encounter this op as it only get generated when lifting Java or
+        // Dalvik bytecode
+        return LiftStatus::kLiftedUnsupportedInstruction;
+      }
+      default: break;
+    }
+
+    return LiftStatus::kLiftedUnsupportedInstruction;
+  }
+
+
+  bool dump(EmittedPcode current_target,
+            std::optional<EmittedPcode> look_ahead_op) override {
+    LOG(INFO) << "inner handle" << std::endl;
+    llvm::IRBuilder bldr(this->target_block);
+
+    // The MULTIEQUAL op has variadic operands
+    if (current_target.opc == OpCode::CPUI_MULTIEQUAL ||
+        current_target.opc == OpCode::CPUI_CPOOLREF) {
+      this->UpdateStatus(
+          this->LiftVariadicOp(bldr, current_target.opc, current_target.outvar,
+                               current_target.vars, current_target.isize),
+          current_target.opc);
+      return false;
+    }
+
+    if (current_target.opc == OpCode::CPUI_CALLOTHER) {
+      if (current_target.isize == 3 &&
+          current_target.vars[0].offset < this->user_op_names.size() &&
```

Comment:
when you are using dot two times to ked to some magic attribute it is a signal abstractions could be leaky.